app/assets/javascripts/tylium/modules/local_auto_save.js

```diff
@@ -1,88 +1,152 @@
+
+      // Check if name is an array, i.e. collection checkboxes
+      if (serializedField.name.slice(-2) == '[]') {
 
+        // When using collection checkboxes, rails/simple_form will create a hidden input with
+        // the same name, i.e. <input type="hidden" name="card[assignee_ids][]" value="">
+        // Don't store that hidden input
+        if (!serializedField.value.length) { return hash; }
+
+        // if array exist, push value to array, else create new array
+        if (hash[serializedField.name]) {
+          hash[serializedField.name].push(serializedField.value);
+        } else {
+          hash[serializedField.name] = [serializedField.value];
         }
-        return hash
+      } else {
+        hash[serializedField.name] = serializedField.value;
       }
 
-      return formInputs.reduce(reducer, {})
+      return hash;
     }
 
-    form.addEventListener("submit", function(event) {
-      if (typeof Storage !== "undefined" && Storage !== null) {
-        localStorage.removeItem(key);
-      }
-    });
+    // serializeArray() is a jquery method that returns an array of objects with key and value attributes.
+    // i.e. [{ name: 'card[name]', value: 1 }, { name: 'card[description]', value: 2 }]
+    return $(this.target).serializeArray().reduce(hashBuilder, {});
+  }
+
+  restoreData() {
+    var data = JSON.parse(localStorage.getItem(this.key));
 
-    document.querySelectorAll("[data-behavior~=clear-local-auto-save]").forEach(function(element) {
-      element.addEventListener("click", function(event) {
-        if (typeof Storage !== "undefined" && Storage !== null) {
-          localStorage.removeItem(key);
+    if (data !== null) {
+      for (let [key, value] of Object.entries(data)) {
+        if (key.slice(-2) == '[]') {
+          value.forEach(function(checkboxValue) {
+            var input = this.target.querySelector(`[name='${key}'][value='${checkboxValue}']`);
+            input.checked = true;
+          })
+        } else {
+          // Query for inputs here first so that we don't query twice
+          var inputs = this.target.querySelectorAll(`[name='${key}']`);
+
+          if (inputs.length) {
+            // Handle checking for radio button and check boxes
+            if (['checkbox', 'radio'].includes(inputs[0].type)) {
+              inputs.forEach(function(input) {
+                if (input.value == value) {
+                  input.checked = true;
+                }
+              });
+            } else {
+              inputs[0].value = value;
+              $(inputs[0]).trigger('load-preview');
+            }
+          }
         }
-      })
-    });
+      }
+    } else {
+      console.log('No data in localStorage for ' + this.key);
+    }
+  }
+}
+
+document.addEventListener('turbolinks:load', function() {
+  document.querySelectorAll('[data-behavior~=local-auto-save]').forEach(function(form) {
+    new LocalAutoSave(form);
   });
 });
```

Comment:
Think it's great you've moved the new local-storage to a module format we've been picking up as a team 👍 

One thing you'll want to do then is remove the initializer here. Have the module file simply define the module. Then in the `behaviors.js` run this loop to initialize the the module on the forms.

Really good refactoring, I'll review the rest in the morning.

Reply:
Thanks, i needed that to initialize the local-auto-save for comments edit.